book/visual-effects.md

```diff
@@ -441,8 +441,8 @@ documentation for more on the Skia API.
 ``` {.python replace=%2c%20scroll/,%20-%20scroll/}
 class DrawLine:
     def execute(self, canvas, scroll):
-        path = skia.Path().moveTo(self.x1 - scroll, self.y1) \
-                          .lineTo(self.x2 - scroll, self.y2)
+        path = skia.Path().moveTo(self.rect.left - scroll, self.rect.top) \
```

Comment:
Missing parentheses in function calls.

Reply:
Your changes to `lab11.py` change from the `Rect` class to `skia.Rect` `DrawRect` (`DrawLine` already uses it); the `Rect` class is no longer used from chapters 11 onward. So it should be `left()` etc.

The book's build system has checks to make sure the chapter's code is exactly the same as the corresponding labNN.py file, which is how I noticed this discrepancy; that's why I recommended `make test` to debug your PR.

Reply:
Please understand my question for a double check, as I'm still not sure.

In my understanding,
Chapter 11 explains the following step by step.

```md
Rasterizing with Skia
=====

(1) ... To draw a line, you use Skia's path object ...
<--- this code here

... similar to draw text using `drawString`;
<-- another diff

(2) ... the `rect` field need to become a skia `Rect` object. ...

class DrawRect:
   ...

(3) Our browser also needs font metrics and measurements. ...

```

In the current flow of Chapter 11,
I'm not sure if you want to introduce the left() method in this code before paragraph (2), which mentions the Skia Rect change.
Similarly, my patch doesn't modify the font measurements and metrics in the code before paragraph (3).

If you'd like, you can fix it instead of my PR.
I'm happy just to report the issue and help you.

Anyway, I found one of my mistakes and fixed it using `make test`

And, I encountered another JS-related error, so I've attached the result.
visual-effects.md looks fine.
[test_result.txt.zip](https://github.com/user-attachments/files/17501771/test_result.txt.zip)

Reply:
Good point. I think the text that talks about switching to `skia.Rect` should move up before `DrawLine` to avoid that. I can make that change after merging your PR, and make then update the website with both changes together.

Thanks for the thorough investigation, I really appreciate it!

Reply:
If you change all of the `left` etc to `left()` now then I can approve the PR. Or if you like I could merge it for you now and then adjust it for you, no problem.

Reply:
> Good point. I think the text that talks about switching to `skia.Rect` should move up before `DrawLine` to avoid that. I can make that change after merging your PR, and make then update the website with both changes together.
> 
> Thanks for the thorough investigation, I really appreciate it!

I absolutely agree with you about that switching.
It will be better to read.

> If you change all of the `left` etc to `left()` now then I can approve the PR. Or if you like I could merge it for you now and then adjust it for you, no problem.

IMHO, I like each commit to be complete, but it doesn't matter in any case.

If you're okay with it, how about accepting this PR and continuing the work?
Alternatively, you can discard this PR and create a new one. I don't mind. :)

Reply:
> If you're okay with it, how about accepting this PR and continuing the work?

I'm definitely okay with it. Will finish reviewing now.